Add new error code to indicate the reset happended in the driver.

pengxin99 opened the pull request:
Use VA_STATUS_ERROR_RESET to indeicate media reset/engine reset have happended in the driver, user can use vaQuerySurfaceError() to get the detail info to do some customized work.

evelikov committed:
va/x11/va_fglrx: remove some dead code

Drop a bunch of set but unused variables and associated dead code.

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>

evelikov committed:
va/x11/va_nvctrl: remove some dead code

Drop a bunch of set but unused variables and associated dead code.

Note: if we are to remove the version handling, then 2/3 of the code in
this file will become unused. Yet it's fairly subtle code and there's a
notable chance of detection breaking if we remove it.

So keep the variables around and (void) prefix them so the compiler does
not warn.

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>

evelikov committed:
va: split the legacy opendriver to separate function

Move all the existing open driver code into a legacy function. It will
soon be replaced (in gradual non-regressing manner) with a new shorter
and more coherent solution.

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>

evelikov-work committed:
va: add vaGetDriverNames() internal ABI

The current GetDriver API is rather fragile and bonkers.

In particular it does a two-step pass - number vs names. That in itself
is not great, since it assumes the devices do not change in between.
Additionally there is the assumption of index X will match device (name)
X, which does not hold true.

The most prominent issue is the inability to provide accurate number and
list of names.

Instead, we can have a single API, that provides both pieces of data in
one go - shorter and simpler code - in reliable manner.

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>

evelikov committed:
drm: implement vaGetDriverNames

v2: Add WSL case

v3: Drop sentinel, use ARRAY_SIZE

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>

evelikov committed:
drm: split DisplayConnect into separate function

This way we can reuse it with the follow-up patches.

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>

evelikov-work committed:
android: split DisplayConnect into separate function

This way we can reuse it with the follow-up patches.

Signed-off-by: Emil Velikov <emil.velikov@collabora.com>

evelikov committed:
drm: remove legacy code paths

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>

evelikov committed:
win32: implement vaGetDriverNames

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>

evelikov committed:
win32: remove legacy code paths

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>

evelikov-work committed:
x11: allow disabling DRI3 via LIBVA_DRI3_DISABLE env var

There are some corner cases where DRI3 does not work correctly. While
bug reports are appreciated, this enables users to get back to their
machines in meaningful way - aka w/o having to rebuild libva.

Signed-off-by: Emil Velikov <emil.velikov@collabora.com>

evelikov-work committed:
android: implement vaGetDriverNames

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>

evelikov-work committed:
android: remove legacy code paths

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>

evelikov-work committed:
wayland: implement vaGetDriverNames

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>

evelikov-work committed:
wayland: remove legacy code paths

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>

evelikov-work committed:
x11: implement vaGetDriverNames

v2: Drop sentinel, use ARRAY_SIZE

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>

evelikov committed:
drm: remove no longer used helpers

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>

evelikov committed:
x11: remove legacy code paths

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>

evelikov committed:
va: remove legacy code paths

Signed-off-by: Emil Velikov <emil.l.velikov@gmail.com>

sivileri committed:
va: Add vendor string on va_TraceInitialize

Signed-off-by: Sil Vilerino <sivileri@microsoft.com>

sivileri committed:
win32: Only print win32 driver messages in DEBUG builds

Closes #694

Signed-off-by: Sil Vilerino <sivileri@microsoft.com>

pengxin99 committed:
Add new VADecodeErrorType to indicate the reset happended in the driver.

Use VADecodeReset to indeicate media reset/engine reset have happended in the driver,
user can use vaQuerySurfaceError() to get the detail info to do some customized work.

Signed-off-by: Pengxin, Yuan <pengxin.yuan@intel.com>

Jexu commented on va/va.h:
```diff
@@ -4215,9 +4223,9 @@ typedef struct _VASurfaceDecodeMBErrors {
- * VA_STATUS_ERROR_DECODING_ERROR should be passed in as "error_status",
+ * After the application gets the VAStatus after calling vaSyncSurface(), and an error may occur,
+ * then need to call vaQuerySurfaceError() to find out further details on the particular VAStatus.
+ * The VAStatus obtained from vaSyncSurface() should be passed in as "error_status",
```
Don't change line4218-4220; Just append some new comments as bellow:

After the applications get VA_STATUS_HW_BUSY or VA_STATUS_SUCCESSFULL from vaSyncSurface, it still can call vaQuerySurfaceError to find out further details to know if has real hw reset happened on this surface since umd and kmd could recover the context from reset with success in sometimes. VA_STATUS_ERROR_RESET should be passed in as 'error_status'.

pengxin99 replied:
Done

evelikov-work committed:
va: don't leak driver names, when override is set

Fixes: 4f5a434 ("va: add vaGetDriverNames() internal ABI")
Signed-off-by: Emil Velikov <emil.velikov@collabora.com>

evelikov-work committed:
va: add missing space in the env.var override info message

Before:
User environment variablerequested driver '...'

After:
User environment variable requested driver '...'

Signed-off-by: Emil Velikov <emil.velikov@collabora.com>

XinfengZhang committed:
update NEWS for 2.19.0

Signed-off-by: Carl Zhang <carl.zhang@intel.com>

XinfengZhang committed:
va: set driver number to be zero if vaGetDriverNames failed

Signed-off-by: Carl Zhang <carl.zhang@intel.com>

XinfengZhang committed:
drm: limit the array size to avoid out of range

map is a array, which could be accessed by index, even the index is out of range.
Fixes #722

Signed-off-by: Carl Zhang <carl.zhang@intel.com>

XinfengZhang committed:
libva 2.20.0.pre1

Signed-off-by: Carl Zhang <carl.zhang@intel.com>
